ibis/expr/api.py

```diff
@@ -643,7 +643,10 @@ def between(arg, lower, upper):
     """
     lower = _ops.as_value_expr(lower)
     upper = _ops.as_value_expr(upper)
-    op = _ops.Between(arg, lower, upper)
+    if isinstance(arg.op(), _ops.Time):
+        op = _ops.BetweenTime(arg, lower, upper)
```

Comment:
We shouldn't need this if we have a `time` datatype, since you'll be able to do:

```python
t.timestamp_col.time().between(lower, upper)
```